src/Post.js

```diff
@@ -37,23 +35,32 @@ class Post extends React.Component {
       .then(this.props.loadPosts)
   }
 
-  handleClick(e) {
+  loadEditForm(e) {
     e.preventDefault();
-    this.setState({
-      editing: true
-    })
-    this.show()
+    this.setState({ editing: true })
   }
 
-  handleSubmit(event) {
-    event.preventDefault();
-    let data = {
-      message: this.state.updated_post
+  fetchParams() {
+    let data = { message: this.state.updated_post }
+    let token = Cookies.get("acebookSession");
+    return {
+      method: 'PUT',
+      mode: 'cors',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        Authorization: token
+      },
+      credentials: 'include',
+      body: JSON.stringify(data)
     }
-    let current_post_id = this.state.post_id
-    updateMessage(data, current_post_id)
+  }
+
+  submitEdit(event) {
+    event.preventDefault();
+    let httpRequest = `${BASE_URL}/api/v1/posts/` + this.props.data.id
```

Comment:
you have called this.props.data.id in the state so would only need this.state.post_id